---
title: "ATAC-seq accessibility exploration"
author: "Koen Van den Berge"
date: "4/1/2021"
output: 
  html_document:
    toc: true
    toc_float: true
---


```{r}
library(ArchR)
archProj <- loadArchRProject("../data/scATAC/archR_subHybridClusters/",
                 showLogo = FALSE)
plotEmbedding(ArchRProj = archProj, colorBy = "cellColData", 
              name = "predAnn", embedding = "UMAP_cor")
archProj <- addMotifAnnotations(ArchRProj = archProj, 
                                motifSet = "cisbp", 
                                name = "Motif",
                                force=TRUE)

## get ArchR peak markers for motif enrichment
markerPeaks_subHybrid <- getMarkerFeatures(
    ArchRProj = archProj, 
    useMatrix = "PeakMatrix", 
    groupBy = "predAnn",
    bias = c("TSSEnrichment", "log10(nFrags)"),
    testMethod = "wilcoxon")
peakMarkers_subHybrid <- getMarkers(markerPeaks_subHybrid, 
                                    cutOff = "FDR <= 0.01 & Log2FC >= 1.25", 
                                    returnGR = TRUE)
heatmapMarkerPeaks <- plotMarkerHeatmap(
  seMarker = markerPeaks_subHybrid, 
  cutOff = "FDR <= 0.01 & Log2FC >= 1.25", 
  transpose = TRUE
)
ComplexHeatmap::draw(heatmapMarkerPeaks, heatmap_legend_side = "bot", annotation_legend_side = "bot")
```

# Check accessibility of Egfr target genes

```{r}
egfrTargetDf <- read.csv("../data/genesTargets/egfr_targets_unfiltered.csv")
pCluster <- plotEmbedding(ArchRProj = archProj, 
                    colorBy = "cellColData", 
                    name = "predAnn", 
                    embedding = "UMAP_cor")
pCluster

allGenes <- unname(unique(mcols(archProj@geneAnnotation$genes)$symbol))
egfrTargets <- intersect(as.character(egfrTargetDf$target), allGenes)

## get peak count matrix
ArrowFiles <- getArrowFiles(archProj)
featureDF <- ArchR:::.getFeatureDF(head(ArrowFiles, 2), "PeakMatrix")
peakMat <- ArchR:::.getPartialMatrix(ArrowFiles, 
            featureDF = featureDF, threads = 1, useMatrix = "PeakMatrix", 
            cellNames = rownames(archProj@cellColData), progress = FALSE)

## for each gene, get the peak closest to TSS
peakGR <- archProj@peakSet
tssPeaks <- c()
for(gg in 1:length(egfrTargets)){
  curGene <- egfrTargets[gg]
  curID <- which(mcols(peakGR)$nearestGene == curGene)
  if(length(curID) > 0 ){
    curTSSPeak <- curID[which.min(mcols(peakGR)$distToTSS[curID])]
    tssPeaks[gg] <- curTSSPeak
  } else {
    tssPeaks[gg] <- NA
    next
  }
}

egfrTargets <- egfrTargets[!is.na(tssPeaks)]
tssPeaks <- tssPeaks[!is.na(tssPeaks)]
egfrTargetPeaks <- peakMat[tssPeaks,]
# egfrTargetPeakScaledSum <- colSums(t(scale(t(egfrTargetPeaks))))
egfrTargetPeakScaledSum <- colSums(egfrTargetPeaks)


umapEmbedding <- getEmbedding(archProj, embedding="UMAP_cor")
colnames(umapEmbedding) <- paste0("UMAP",1:2)
umapEmbedding$egfrTargetScaledSum <- egfrTargetPeakScaledSum

pEgfrTargets <- ggplot(umapEmbedding, aes(x=UMAP1,
                          y=UMAP2,
                          col=egfrTargetScaledSum)) +
  theme_classic() +
  geom_point() +
  scale_color_gradientn(colors=wesanderson::wes_palette("Zissou1", n=100, type="continuous"))

pEgfrTargets
```

# Check accessibility of peaks containining Ets TFBS motif

```{r}
motifMatches <- readRDS(archProj@peakAnnotation$Motif$Matches)
etsMotifId <- grep(x=colnames(motifMatches), pattern="Ets")
etsMotifs <- grep(x=colnames(motifMatches), pattern="Ets", value=TRUE)
peaksWithEtsMotifs <- which(rowSums(assays(motifMatches[,etsMotifs])$matches) > 0)
length(peaksWithEtsMotifs)

umapEmbedding$etsMotifPeaks <- colSums(peakMat[peaksWithEtsMotifs,])

pEtsMotifs <- ggplot(umapEmbedding, aes(x=UMAP1,
                          y=UMAP2,
                          col=etsMotifPeaks)) +
  theme_classic() +
  geom_point() +
  scale_color_gradientn(colors=wesanderson::wes_palette("Zissou1", n=100, type="continuous"))

pEtsMotifs
```

# Association between Egfr targets' accessibility and Ets motif peaks' accessibility

```{r}
plot(umapEmbedding$egfrTargetScaledSum, umapEmbedding$etsMotifPeaks,
     xlab = "Egfr targets' accessibility",
     ylab = "Ets-motif-containing peaks' accessibility")
```

# Gene markers for (sub)hybrid clusters

```{r}
plotEmbedding(ArchRProj = archProj, colorBy = "cellColData", 
              name = "predAnn", embedding = "UMAP_cor")
```


```{r}
## get ArchR gene markers for motif enrichment
markerGenes_subHybrid <- getMarkerFeatures(
    ArchRProj = archProj, 
    useMatrix = "GeneScoreMatrix", 
    groupBy = "predAnn",
    bias = c("TSSEnrichment", "log10(nFrags)"),
    testMethod = "wilcoxon")
geneMarkers_subHybrid <- getMarkers(markerGenes_subHybrid, 
                                    cutOff = "FDR <= 0.01 & Log2FC >= 1.25")

write.csv(geneMarkers_subHybrid$C2_Hybrid, file="../data/scATAC/markerGenesHybrid.csv")
write.csv(geneMarkers_subHybrid$hybrid1, file="../data/scATAC/markerGenesHybrid1.csv")
write.csv(geneMarkers_subHybrid$hybrid2, file="../data/scATAC/markerGenesHybrid2.csv")
write.csv(geneMarkers_subHybrid$C1_Inj, file="../data/scATAC/markerGenesInjured.csv")
write.csv(geneMarkers_subHybrid$resting, file="../data/scATAC/markerGenesResting.csv")
```

# Marker genes for hybrid1

```{r}
markerGenesHybrid1 <- geneMarkers_subHybrid$hybrid1$name[1:18]
p <- plotEmbedding(
    ArchRProj = archProj, 
    colorBy = "GeneScoreMatrix", 
    name = markerGenesHybrid1, 
    embedding = "UMAP_cor",
    quantCut = c(0.01, 0.95),
    size=1/3,
    plotAs="points"
)
# clean
p2Hybrid1 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
cowplot::plot_grid(plotlist=p2Hybrid1[1:9])
cowplot::plot_grid(plotlist=p2Hybrid1[10:18])

```

## Injured/uninjured contribution of marker genes

```{r}
geneScoreMat <- ArchR::getMatrixFromProject(archProj,
                            useMatrix = "GeneScoreMatrix")
geneScoreMatHybrid1 <- geneScoreMat[,colData(geneScoreMat)$predAnn == "hybrid1"]

table(geneScoreMatHybrid1$treat)

rafalib::mypar(mfrow=c(3,3))
for(gg in 1:length(markerGenesHybrid1)){
  boxplot(log1p(assays(geneScoreMatHybrid1)$GeneScoreMatrix[which(rowData(geneScoreMatHybrid1)$name == markerGenesHybrid1[gg]),]) ~ geneScoreMatHybrid1$treat,
        xlab = "Treatment", ylab = "Log Gene activity score + 1",
        main = markerGenesHybrid1[gg])
}
```



# Marker genes for hybrid2

```{r}
markerGenesHybrid2 <- geneMarkers_subHybrid$hybrid2$name[1:18]
p <- plotEmbedding(
    ArchRProj = archProj, 
    colorBy = "GeneScoreMatrix", 
    name = markerGenesHybrid2, 
    embedding = "UMAP_cor",
    quantCut = c(0.01, 0.95),
    size=1/3,
    plotAs="points"
)
# clean
p2Hybrid2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
cowplot::plot_grid(plotlist=p2Hybrid2[1:9])
cowplot::plot_grid(plotlist=p2Hybrid2[10:18])

```

## Injured/uninjured contribution of marker genes

```{r}
geneScoreMatHybrid2 <- geneScoreMat[,colData(geneScoreMat)$predAnn == "hybrid2"]

table(geneScoreMatHybrid2$treat)

rafalib::mypar(mfrow=c(3,3))
for(gg in 1:length(markerGenesHybrid2)){
  boxplot(log1p(assays(geneScoreMatHybrid2)$GeneScoreMatrix[which(rowData(geneScoreMatHybrid2)$name == markerGenesHybrid2[gg]),]) ~ geneScoreMatHybrid2$treat,
        xlab = "Treatment", ylab = "Log Gene activity score + 1",
        main = markerGenesHybrid2[gg])
}

```

# Marker genes for other hybrid cells

```{r}
markerGenesHybrid <- geneMarkers_subHybrid$C2_Hybrid$name[1:18]
p <- plotEmbedding(
    ArchRProj = archProj, 
    colorBy = "GeneScoreMatrix", 
    name = markerGenesHybrid, 
    embedding = "UMAP_cor",
    quantCut = c(0.01, 0.95),
    size=1/3,
    plotAs="points"
)
# clean
p2Hybrid <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
cowplot::plot_grid(plotlist=p2Hybrid[1:9])
cowplot::plot_grid(plotlist=p2Hybrid[10:18])

```


# Accesssibility of key lineage-specific genes from other cell types / lineages

```{r}
# John suggested Kit1, but there's only Kit and Kitl
# 
markerGenesLineageSpecific <- c("Sox2", "Ascl1", "Kit", "Neurog1", "Neurod1")
p <- plotEmbedding(
    ArchRProj = archProj, 
    colorBy = "GeneScoreMatrix", 
    name = markerGenesLineageSpecific, 
    embedding = "UMAP_cor",
    quantCut = c(0.01, 0.95),
    size=1/3,
    plotAs="points"
)
# clean
p2LineageSpecific <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
cowplot::plot_grid(plotlist=p2LineageSpecific)
```

