---
title: "OE Regeneration Main - Figure 8"
author: "Divya Kunda"
date: '12/5/2022'
output: html_document
---

```{r pkg}
#load packages
suppressPackageStartupMessages({
  library(slingshot)
  library(SingleCellExperiment)
  library(ggplot2)
  library(parallel)
  library(ArchR)
  library(plyr)
  # library(ggbreak)
  library(tradeSeq)
})
```

# Import data

```{r loadData}
archProj <- loadArchRProject("../data/scATAC/archR_subHybridClusters/", showLogo = FALSE)
sds <- readRDS("../data/finalTrajectory/sling.rds")
counts <- readRDS("../data/finalTrajectory/counts_noResp_noMV.rds")
counts <- round(counts)
sce <- readRDS("../data/finalTrajectory/sce_tradeSeq20200904.rds")
load("../data/ALL_TF.Rda")
clDatta <- readRDS("../data/finalTrajectory/dattaCl_noResp_noMV.rds")
timePoint <- readRDS("../data/finalTrajectory/timePoint_noResp_noMV.rds")
```

# scATAC-seq analysis of lineage-specific genes

```{r}
lsGenes <- c("Cd36", "Aldh1a7", "Trim46", "Erich3", "Cap2", "Adh7")

pLS <- plotEmbedding(
  ArchRProj = archProj, 
  colorBy = "GeneScoreMatrix", 
  name = lsGenes, 
  embedding = "UMAP_cor",
  quantCut = c(0.01, 0.95),
  size=1,
  plotAs="points"
)

#clean
p2LS <- lapply(pLS, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})

pLS
```

## Density plots of gene activity scores

```{r}
geneScoreMat <- ArchR::getMatrixFromProject(archProj, useMatrix="GeneScoreMatrix")

dfLS <- data.frame(
  Cd36Acc = assays(geneScoreMat)$GeneScoreMatrix[which(rowData(geneScoreMat)$name == "Cd36"),],
  Aldh1a7Acc = assays(geneScoreMat)$GeneScoreMatrix[which(rowData(geneScoreMat)$name == "Aldh1a7"),],
  Trim46Acc = assays(geneScoreMat)$GeneScoreMatrix[which(rowData(geneScoreMat)$name == "Trim46"),],
  Erich3Acc = assays(geneScoreMat)$GeneScoreMatrix[which(rowData(geneScoreMat)$name == "Erich3"),],
  Cap2Acc = assays(geneScoreMat)$GeneScoreMatrix[which(rowData(geneScoreMat)$name == "Cap2"),],
  Adh7Acc = assays(geneScoreMat)$GeneScoreMatrix[which(rowData(geneScoreMat)$name == "Adh7"),], 
  ct = geneScoreMat@colData$clHBCMerged, 
  treat = geneScoreMat@colData$treat)

```

### Cd36

```{r}
phi <- ddply(dfLS, "treat", summarise, treat.median = median(log1p(Cd36Acc)))
mu <- ddply(dfLS, "treat", summarise, treat.mean = mean(log1p(Cd36Acc)))

densCd36_hyb <- ggplot(dfLS[dfLS$ct == 'C2_Hybrid',], aes(x = log1p(Cd36Acc), col = treat, fill = treat)) +
  ggtitle('gene activity in hybrid cluster') +
  theme_classic() +
  theme(legend.position = 'none',
        axis.line = element_line(size = .5), 
        axis.ticks = element_line(size = .5), 
        axis.text = element_text(size = 12)) +
  geom_density(alpha=0, size = .75) +
  scale_colour_manual(values = c("orange", "darkseagreen3")) +
  geom_vline(data = phi, aes(xintercept = treat.median, color = treat), linetype = "dashed", size = .75) 

densCd36_hyb
phi
mu
```

### Aldh1a7

```{r}
phi <- ddply(dfLS, "treat", summarise, treat.median = median(log1p(Aldh1a7Acc)))
mu <- ddply(dfLS, "treat", summarise, treat.mean = mean(log1p(Aldh1a7Acc)))

densAldh1a7_hyb <- ggplot(dfLS[dfLS$ct == 'C2_Hybrid',], aes(x = log1p(Aldh1a7Acc), col = treat, fill = treat)) +
  ggtitle('gene activity in hybrid cluster') +
  theme_classic() +
  theme(legend.position = 'none',
        axis.line = element_line(size = .5), 
        axis.ticks = element_line(size = .5), 
        axis.text = element_text(size = 12)) +
  geom_density(alpha=0, size = .75) +
  scale_colour_manual(values = c("orange", "darkseagreen3")) +
  geom_vline(data = phi, aes(xintercept = treat.median, color = treat), linetype = "dashed", size = .75) +
  scale_y_break(c(1.5, 4), scales = 0.5) 

densAldh1a7_hyb
phi
mu
```

### Trim46

```{r}
phi <- ddply(dfLS, "treat", summarise, treat.median = median(log1p(Trim46Acc)))
mu <- ddply(dfLS, "treat", summarise, treat.mean = mean(log1p(Trim46Acc)))

densTrim46_hyb <- ggplot(dfLS[dfLS$ct == 'C2_Hybrid',], aes(x = log1p(Trim46Acc), col = treat, fill = treat)) +
  ggtitle('gene activity in hybrid cluster') +
  theme_classic() +
  theme(legend.position = 'none',
        axis.line = element_line(size = .5), 
        axis.ticks = element_line(size = .5), 
        axis.text = element_text(size = 12)) +
  geom_density(alpha=0, size = .75) +
  scale_colour_manual(values = c("orange", "darkseagreen3")) +
  geom_vline(data = phi, aes(xintercept = treat.median, color = treat), linetype = "dashed", size = .75) 

densTrim46_hyb
phi
mu
```

### Erich3

```{r}
phi <- ddply(dfLS, "treat", summarise, treat.median = median(log1p(Erich3Acc)))
mu <- ddply(dfLS, "treat", summarise, treat.mean = mean(log1p(Erich3Acc)))

densErich3_hyb <- ggplot(dfLS[dfLS$ct == 'C2_Hybrid',], aes(x = log1p(Erich3Acc), col = treat, fill = treat)) +
  ggtitle('gene activity in hybrid cluster') +
  theme_classic() +
  theme(legend.position = 'none',
        axis.line = element_line(size = .5), 
        axis.ticks = element_line(size = .5), 
        axis.text = element_text(size = 12)) +
  geom_density(alpha=0, size = .75) +
  scale_colour_manual(values = c("orange", "darkseagreen3")) +
  geom_vline(data = phi, aes(xintercept = treat.median, color = treat), linetype = "dashed", size = .75) +
  scale_y_break(c(2, 3), scales = 0.5) 

densErich3_hyb
phi
mu
```

### Cap2

```{r}
phi <- ddply(dfLS, "treat", summarise, treat.median = median(log1p(Cap2Acc)))
mu <- ddply(dfLS, "treat", summarise, treat.mean = mean(log1p(Cap2Acc)))

densCap2_hyb <- ggplot(dfLS[dfLS$ct == 'C2_Hybrid',], aes(x = log1p(Cap2Acc), col = treat, fill = treat)) +
  ggtitle('gene activity in hybrid cluster') +
  theme_classic() +
  theme(legend.position = 'none',
        axis.line = element_line(size = .5), 
        axis.ticks = element_line(size = .5), 
        axis.text = element_text(size = 12)) +
  geom_density(alpha=0, size = .75) +
  scale_colour_manual(values = c("orange", "darkseagreen3")) +
  geom_vline(data = phi, aes(xintercept = treat.median, color = treat), linetype = "dashed", size = .75) 

densCap2_hyb
phi
mu
```

### Adh7

```{r}
phi <- ddply(dfLS, "treat", summarise, treat.median = median(log1p(Adh7Acc)))
mu <- ddply(dfLS, "treat", summarise, treat.mean = mean(log1p(Adh7Acc)))

densAdh7_hyb <- ggplot(dfLS[dfLS$ct == 'C2_Hybrid',], aes(x = log1p(Adh7Acc), col = treat, fill = treat)) +
  ggtitle('gene activity in hybrid cluster') +
  theme_classic() +
  theme(legend.position = 'none',
        axis.line = element_line(size = .5), 
        axis.ticks = element_line(size = .5), 
        axis.text = element_text(size = 12)) +
  geom_density(alpha=0, size = .75) +
  scale_colour_manual(values = c("orange", "darkseagreen3")) +
  geom_vline(data = phi, aes(xintercept = treat.median, color = treat), linetype = "dashed", size = .75) 

densAdh7_hyb
phi
mu
```

## scRNA-seq trajectory inference

```{r}
labels <- c("Neur", "Sus", "rHBC")
names(labels) <- as.character(1:3)
```

### Cd36

```{r}
trCd36 <- plotSmoothers(sce, counts, gene="Cd36") + 
  scale_color_viridis_d(labels=labels) + 
  ggtitle("Cd36") +
  theme(legend.position = "none")

trCd36
```

### Aldh1a7

```{r}
trAldh1a7 <- plotSmoothers(sce, counts, gene="Aldh1a7") + 
  scale_color_viridis_d(labels=labels) + 
  ggtitle("Aldh1a7") +
  theme(legend.position = "none")

trAldh1a7
```

### Trim46

```{r}
trTrim46 <- plotSmoothers(sce, counts, gene="Trim46") + 
  scale_color_viridis_d(labels=labels) + 
  ggtitle("Trim46") +
  theme(legend.position = "none")

trTrim46
```

### Erich3

```{r}
trErich3 <- plotSmoothers(sce, counts, gene="Erich3") + 
  scale_color_viridis_d(labels=labels) + 
  ggtitle("Erich3") +
  theme(legend.position = "none")

trErich3
```

### Cap2

```{r}
trCap2 <- plotSmoothers(sce, counts, gene="Cap2") + 
  scale_color_viridis_d(labels=labels) + 
  ggtitle("Cap2") +
  theme(legend.position = "none")

trCap2
```

### Adh7

```{r}
trAdh7 <- plotSmoothers(sce, counts, gene="Adh7") + 
  scale_color_viridis_d(labels=labels) + 
  ggtitle("Adh7") +
  theme(legend.position = "none")

trAdh7
```

